Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -644,6 +645,23 @@ public override void Reset(VCExpressionGenerator gen)
       }
     }
 
+    private void RecoverIfProverCrashedAfterReset()
+    {
+      resetCount += 1;
+      if (Process.HasProverDied() is { } e)
+      {
+        if (resetCount == 1)
+        {
+          // First-time reset cannot be recovered
```

Comment:
Why not?

Reply:
Because at this point, only `(reset)` was sent to the solver. if it crashes there, recovering it would mean spawning a new solver instance and sending it more `(reset)` command, which could loop indefinitely.
In practice this then-branch will never be entered, but it's there to prove termination (in the unlikely case a solver is corrupted for example)

Reply:
Why would another reset command be sent after the recovery?

Reply:
You're right, actually, no `(reset)` command is sent after the recover. I think I can simplify my code then.